src/tty.rs

```diff
@@ -186,9 +186,20 @@ pub fn new<T: ToWinsize>(config: &Config, size: T) -> Pty {
 
     let (master, slave) = openpty(win.ws_row as _, win.ws_col as _);
 
-    let shell = config.shell().unwrap_or(pw.shell);
+    let (program, args) = match config.shell() {
```

Comment:
oh that would be a type mismatch... We'll just have to initialize the default shell before choosing one. Assuming you add a `let default_shell = Shell::new(pw.shell)`, The default becomes `unwrap_or(&default_shell)`.

I _think_ that should type check now.